src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -77,22 +81,72 @@ private void AcceptFailed(SocketError error, string? exceptionMessage)
         }
 
         [NonEvent]
-        public void ConnectStart(SocketAddress address)
+        public Activity? ConnectStart(SocketAddress address, EndPoint endPoint, bool keepActivityCurrent)
         {
             Interlocked.Increment(ref _currentOutgoingConnectAttempts);
 
             if (IsEnabled(EventLevel.Informational, EventKeywords.All))
             {
                 ConnectStart(address.ToString());
             }
+
+            Activity? activity = null;
+            if (s_connectActivitySource.HasListeners())
+            {
+                Activity? activityToReset = keepActivityCurrent ? Activity.Current : null;
+                activity = s_connectActivitySource.StartActivity(ConnectActivityName);
+                if (keepActivityCurrent)
+                {
+                    // Do not overwrite Activity.Current in the caller's ExecutionContext.
+                    Activity.Current = activityToReset;
+                }
+            }
+
+            if (activity is not null)
+            {
+                if (endPoint is IPEndPoint ipEndPoint)
+                {
+                    int port = ipEndPoint.Port;
+                    activity.DisplayName = $"socket connect {ipEndPoint.Address}:{port}";
+                    if (activity.IsAllDataRequested)
+                    {
+                        activity.SetTag("network.peer.address", ipEndPoint.Address.ToString());
+                        activity.SetTag("network.peer.port", port);
+                        activity.SetTag("network.type", ipEndPoint.AddressFamily == AddressFamily.InterNetwork ? "ipv4" : "ipv6");
+                    }
```

Comment:
wdyt about https://github.com/open-telemetry/semantic-conventions/pull/1192#discussion_r1673448739 ? 
adding `network.transport` if it's `udp` or `unix`

Reply:
This would also make sense for QUIC in the future.

Reply:
UDS doesn't map well to the OSI semantics used by the [`network.*`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/network/) attributes. If anything, it's rather in the network than in the transport layer: sockets are created with `AddressFamily=Unix`, `ProtocolType=Unspecified`.

From .NET perspective "QUIC is not sockets": we should not emit Activities from the same `System.Net.Sockets` activity span. Also, mapping QUIC to OSI would mean that `network.type=ipv4/6` and `network.transport=udp`, which is useless from diagnostics perspective.

IMO semconv has [some homework](https://github.com/open-telemetry/semantic-conventions/issues/1237) to do here, until then I prefer to avoid emitting stuff based on assumptions. My prefrence would be to migrate away from the archaic OSI semantics, it's unfortunate that the related `network.*` attributes are marked as stable already.

Reply:
semconv tell to set `network.transport=unix` 

https://github.com/open-telemetry/semantic-conventions/blob/f17a4bf973d32a0888a8219a408397f8eeac12ed/docs/attributes-registry/network.md?plain=1#L96

I agree UDS does not map well to OSI, so whatever we did would be imperfect, but this ship has sailed, it's stable

Reply:
Sorry, missed that one! QUIC is still unclear though :)

Reply:
there seem to be is no controversy in adding quic as one of the values of `network.transport` - I replied on the semconv issue. It fits into the OSI levels and the attribute definition perfectly. 
The `network.transport` is an enum and all enums in semconv are open - so it should be fine to add it without it being mentioned explicitly.

> `network.transport` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used

I can send the PR to semconv to add it, but it'd probably takes a few days to merge.